frontend/views/containers/chatroom/DMMixin.js

```diff
@@ -11,27 +11,28 @@ const DMMixin: Object = {
       'ourGroupDirectMessages',
       'ourIdentityContractId',
       'ourGroupDirectMessageFromUserIds'
-    ])
+    ]),
+    ...mapState(['currentGroupId'])
   },
   methods: {
     async createDirectMessage (memberIDs: string | string[]) {
       if (typeof memberIDs === 'string') {
         memberIDs = [memberIDs]
       }
       try {
+        const identityContractID = this.ourIdentityContractId
+        const currentGroupId = this.currentGroupId
         await sbp('gi.actions/identity/createDirectMessage', {
-          contractID: this.ourIdentityContractId,
-          data: { memberIDs },
+          contractID: identityContractID,
+          data: { currentGroupId, memberIDs },
           hooks: {
-            onprocessed: (message) => {
-              const dmID = message.decryptedValue().data.contractID
-              // The logic for updating paths will not work until the DM chatroom
-              // has been synced
-              sbp('chelonia/queueInvocation', dmID, () => this.redirect(dmID))
+            prepublish (message) {
+              sbp('state/vuex/commit', 'setPendingChatRoomId', { chatRoomID: message.contractID(), groupID: currentGroupId })
```

Comment:
What's this change about? 🤔 

Very different stuff going on all of a sudden. Maybe also needs a comment.

Reply:
I can add an explanation. The basic idea is that there's an interval of time where the chatroom ID is known but not yet synced. Trying to set the chatroom ID will cause weird behaviour, including the `alert` that it's a private chatroom.